core/src/test/kotlin/io/github/kkafka/producer/TransactionKtTest.kt

```diff
@@ -0,0 +1,191 @@
+package io.github.kkafka.producer
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.errors.TransactionAbortedException
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+internal class TransactionKtTest {
+    private val topic = "kf-some-topic"
+    private val topics = setOf(topic, "kf-some-other-topic")
+
+    private val consumer: KafkaConsumer<Int, String> = mockk {
+        every { assignment() } answers {
+            // Add a topic we don't care about to make sure it is ignored
+            val allTopics = topics + setOf("kf-some-uninteresting-topic")
+            allTopics.map { TopicPartition(it, 0) }.toSet()
+        }
+        every { groupMetadata().groupId() } answers { "someGroupId" }
+        every { position(any()) } answers { 31 }
+    }
+
+    private val producer: KafkaProducer<String, String> = mockk {
+        every { beginTransaction() } answers {}
```

Comment:
for future reference: if you want functions that return Unit to be mocked, you can also write 
````kotlin
every {  /* ... */ } just runs
````


or 

````kotlin
private val consumer: KafkaConsumer<Int, String> = mockk(relaxUnitFun = true) {
    // ...
}
````